squirrel.frontier/src/main/java/org/dice_research/squirrel/queue/scorecalculator/URIGraphSizeBasedScoreCalculator.java

```diff
@@ -0,0 +1,109 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(URIGraphSizeBasedScoreCalculator.class);
+
+    protected QueryExecutionFactory queryExecFactory = null;
+
+    public URIGraphSizeBasedScoreCalculator() {
+        throw new UnsupportedOperationException();
+    }
+
+    public URIGraphSizeBasedScoreCalculator(QueryExecutionFactory qe) {
+        this.queryExecFactory = qe;
+    }
+
+    public URIGraphSizeBasedScoreCalculator(String sparqlEndpointUrl, String username, String password) {
+        if (username != null && password != null) {
+            // Create the factory with the credentials
+            final Credentials credentials = new UsernamePasswordCredentials(username, password);
+            HttpAuthenticator authenticator = new HttpAuthenticator() {
+                @Override
+                public void invalidate() {
+                    // unused method in this implementation
+                }
+
+                @Override
+                public void apply(AbstractHttpClient client, HttpContext httpContext, URI target) {
+                    client.setCredentialsProvider(new CredentialsProvider() {
+                        @Override
+                        public void clear() {
+                            // unused method in this implementation
+                        }
+
+                        @Override
+                        public Credentials getCredentials(AuthScope scope) {
+                            return credentials;
+                        }
+
+                        @Override
+                        public void setCredentials(AuthScope arg0, Credentials arg1) {
+                            LOGGER.error("I am a read-only credential provider but got a call to set credentials.");
+                        }
+                    });
+                }
+            };
+            this.queryExecFactory = new QueryExecutionFactoryHttp(sparqlEndpointUrl, new DatasetDescription(), authenticator);
+        } else {
+            this.queryExecFactory = new QueryExecutionFactoryHttp(sparqlEndpointUrl);
+        }
+
+    }
+
+    /**
+     * This method calculates the duplicity score for a {@link CrawleableUri} which is 1 / (number of times the Uri
+     * occurs as a subject in graphs).
+     *
+     * @param uri the {@link CrawleableUri} for which the duplicity score has to be calculated
+     * @return duplicity score
+     */
+    public float getURIScore(CrawleableUri uri) {
+        int uriScore = getGraphSize(uri.getUri().toString());
+        if (uriScore == 0) {
+            return 1;
+        }
+        return 1 / (float) uriScore;
+    }
+
+    /**
+     * This method return the number of times the Uri occurs as a subject in the graphs.
+     *
+     * @param uri the Uri for which the score has to calculated
+     * @return number of times the Uri occurs as a subject in the graphs
+     */
+    private int getGraphSize(String uri) {
```

Comment:
"This method return the number of times the Uri occurs as a subject in the graphs." -> In that case the method (and the class) name does not really fit. Maybe you can come up with a better name, e.g., getSubjectTripleCount or something similar.